src/sys/reboot.rs

```diff
@@ -1,48 +1,163 @@
+                RB_AUTOBOOT,
+                // we do not support Restart2.
+                /// Suspend the system using software suspend.
+                RB_SW_SUSPEND,
+            }
+        }
+
+        /// Reboots or shuts down the system.
+        #[cfg(target_os = "linux")]
+        pub fn reboot(how: RebootMode) -> Result<Infallible> {
+            unsafe { libc::reboot(how as libc::c_int) };
+            Err(Errno::last())
+        }
 
-/// Enable or disable the reboot keystroke (Ctrl-Alt-Delete).
-///
-/// Corresponds to calling `reboot(RB_ENABLE_CAD)` or `reboot(RB_DISABLE_CAD)` in C.
-pub fn set_cad_enabled(enable: bool) -> Result<()> {
-    let cmd = if enable {
-        libc::RB_ENABLE_CAD
-    } else {
-        libc::RB_DISABLE_CAD
-    };
-    let res = unsafe { libc::reboot(cmd) };
-    Errno::result(res).map(drop)
+        /// Enable or disable the reboot keystroke (Ctrl-Alt-Delete).
+        ///
+        /// Corresponds to calling `reboot(RB_ENABLE_CAD)` or `reboot(RB_DISABLE_CAD)` in C.
+        #[cfg(target_os = "linux")]
+        pub fn set_cad_enabled(enable: bool) -> Result<()> {
+            let cmd = if enable {
+                libc::RB_ENABLE_CAD
+            } else {
+                libc::RB_DISABLE_CAD
+            };
+            let res = unsafe { libc::reboot(cmd) };
+            Errno::result(res).map(drop)
+        }
+    } else if #[cfg(netbsdlike)] {
+        #[cfg(target_os = "netbsd")]
+        use std::os::unix::ffi::OsStrExt;
+        #[cfg(target_os = "netbsd")]
+        use std::ffi::OsStr;
+        use libc::c_int;
+
+        libc_bitflags! {
+            /// How exactly should the system be rebooted.
+            pub struct RebootMode: c_int {
+                /// The default, causing the system to reboot in its usual fashion.
+                RB_AUTOBOOT;
+                /// Interpreted by the bootstrap program itself, causing it to
+                /// prompt on the console as to what file should be booted.
+                /// Normally, the system is booted from the file “xx(0,0)bsd”,
+                /// where xx is the default disk name, without prompting for
+                /// the file name.
+                RB_ASKNAME;
+                /// Dump kernel memory before rebooting; see `savecore(8)` for
+                /// more information.
+                RB_DUMP;
+                /// The processor is simply halted; no reboot takes place.
+                RB_HALT;
+                /// Power off the system if the system hardware supports the
+                /// function, otherwise it has no effect.
+                ///
+                /// Should be used in conjunction with `RB_HALT`.
+                RB_POWERDOWN;
+                /// An option allowing the specification of an init program (see
+                /// `init(8)`) other than `/sbin/init` to be run when the system
+                /// reboots.
+                ///
+                /// This switch is not currently available.
```

Comment:
What does it mean that it isn't currently available?  Does that mean that NetBSD defines it but does not implement it?  If so, I think that Nix shouldn't expose it.

Reply:
From this [man page](https://man.netbsd.org/NetBSD-8.0/boothowto.9), seems that it is obsolete and replaced with `RB_ASKNAME`, removed

> RB_INITNAME     0x00000010          This flag is obsolete.  It was previously used to cause the kernel to prompt for the name of the [init(8)](https://man.netbsd.org/NetBSD-8.0/init.8) program, but that function is now controlled by the RB_ASKNAME flag.